File: github_hooks/db/migrate/20251119103829_add_expires_created_index_at_jobs_table.rb
class AddExpiresCreatedIndexAtJobsTable < ActiveRecord::Migration[6.1]
  disable_ddl_transaction!

  def change
    add_index :jobs,
              :if_not_exists,
              %i[expires_at created_at],
              name: "index_jobs_on_expires_created_not_null",
              algorithm: :concurrently,
              where: "expires_at IS NOT NULL"
  end
end

File: zebra/priv/legacy_repo/migrations/20251119103829_add_expires_created_index_at_jobs_table.exs
defmodule Zebra.LegacyRepo.Migrations.AddExpiresCreatedIndexAtJobsTable do
  use Ecto.Migration
  @disable_migration_lock true
  @disable_ddl_transaction true

  def change do
    create index(:jobs, [:expires_at, :created_at],
      name: "index_jobs_on_expires_created_not_null",
      concurrently: true,
      where: "expires_at IS NOT NULL",
      if_not_exists: true
    )
  end
end
